engine/src/main/java/org/terasology/rendering/world/WorldRendererLwjgl.java

```diff
@@ -649,23 +644,17 @@ private void renderChunk(RenderableChunk chunk, ChunkMesh.RenderPhase phase, Cam
                         statRenderedTriangles += 12;
                     }
 
-                    if (shader != null) {
-                        shader.enable();
-                    }
```

Comment:
This is the block that caused the Icon Atlas to show up instead of the water surface. Specifically, in the faulty commit I thought I could remove it. It turns out it's possible but not without moving the chunkShader.enable() command at the end of the if(mode || mode) block it is part of. 

This seems to be due to a lot of shader enabling/disabling happening behind the scene when shader features are activated or deactivated.